src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotUseInsecureSettingsForJsonNetTests.cs

```diff
@@ -0,0 +1,528 @@
+        }
+
+        [Fact]
+        public void Secure_JsonSerializer_CreateDefault_NoDiagnostic()
+        {
+            this.VerifyCSharpWithJsonNet(@"
+using System.IO;
+using Newtonsoft.Json;
+
+class Blah
+{
+    T Deserialize<T>(string s)
+    {
+        JsonSerializerSettings settings = new JsonSerializerSettings();
+        JsonSerializer serializer = JsonSerializer.Create(settings);
+        return (T) serializer.Deserialize(new StringReader(s), typeof(T));
+    }
+}");
+        }
+
+        [Fact]
+        public void Insecure_JsonConvert_DefaultSettings_Lambda_DefinitelyDiagnostic()
+        {
+            this.VerifyCSharpWithJsonNet(@"
+using Newtonsoft.Json;
+
+class Blah
+{
+    void Method()
+    {
+        JsonConvert.DefaultSettings = () =>
+        {
+            JsonSerializerSettings settings = new JsonSerializerSettings();
+            settings.TypeNameHandling = TypeNameHandling.All;
+            return settings;
+        };
+    }
+}",
+                GetCSharpResultAt(12, 20, DefinitelyRule));
+        }
+
+        [Fact]
+        public void Insecure_JsonConvert_DefaultSettings_Lambda_ImplicitReturn_DefinitelyDiagnostic()
+        {
+            this.VerifyCSharpWithJsonNet(@"
+using Newtonsoft.Json;
+
+class Blah
+{
+    void Method()
+    {
+        JsonConvert.DefaultSettings = () =>
+            new JsonSerializerSettings()
+            {
+                TypeNameHandling = TypeNameHandling.All,
+            };
+    }
+}",
+                GetCSharpResultAt(9, 13, DefinitelyRule));
+        }
+
+        [Fact]
+        public void Insecure_JsonConvert_DefaultSettings_LocalFunction_DefinitelyDiagnostic()
+        {
+            this.VerifyCSharpWithJsonNet(@"
+using Newtonsoft.Json;
+
+class Blah
+{
+    void Method()
+    {
+        JsonConvert.DefaultSettings = GetSettings;
+
+        JsonSerializerSettings GetSettings()
+        {
+            JsonSerializerSettings settings = new JsonSerializerSettings();
+            settings.TypeNameHandling = TypeNameHandling.All;
+            return settings;
+        };
+    }
+}",
+                GetCSharpResultAt(14, 20, DefinitelyRule));
+        }
+
+        [Fact]
+        public void Insecure_JsonConvert_DefaultSettings_LocalFunctionWithTryCatch_DefinitelyDiagnostic()
+        {
+            this.VerifyCSharpWithJsonNet(@"
+using System;
+using Newtonsoft.Json;
+
+class Blah
+{
+    void Method()
+    {
+        JsonConvert.DefaultSettings = GetSettings;
+
+        JsonSerializerSettings GetSettings()
+        {
+            JsonSerializerSettings settings = new JsonSerializerSettings();
+            try
+            {
+                settings.TypeNameHandling = TypeNameHandling.All;
+            }
+            catch (Exception ex)
+            {
+                HandleException(ex);
+            }
+
+            return settings;
+        };
+    }
+
+    void HandleException(Exception exParam)  // 'ex' asserts in AnalysisEntityFactory.EnsureLocation()
```

Comment:
>'ex' asserts in AnalysisEntityFactory.EnsureLocation() [](start = 48, length = 54)

@mavasani have any ideas off the top of your head, before I dive too deep into this?